src/main/java/io/cdap/plugin/http/source/common/pagination/IncrementAnIndexPaginationIterator.java

```diff
@@ -61,4 +67,28 @@ protected String getNextPageUrl(HttpResponse response, BasePage page) {
   public boolean supportsSkippingPages() {
     return true;
   }
+
+  @Override
+  public void setNextPageUrl(String nextPageUrl) {
+    super.setNextPageUrl(nextPageUrl);
+
+    Matcher matcher = urlPatten.matcher(nextPageUrl);
+    if (!matcher.matches()) {
+      throw new IllegalArgumentException(String.format(
+        "Cannot set nextPageUrl to '%s' as it does not match the original url '%s'", nextPageUrl, config.getUrl()));
```

Comment:
Simple English text no nextPageUrl variable names. Refer to display names

Reply:
Done. Actually code with exception got removed due to changes suggested by Albert.